HW_11/Task_1_2.js

```diff
@@ -0,0 +1,194 @@
+// 7. Добавьте методы:
+//     addEmployee(employee) — добавляет сотрудника в массив employees с проверкой, что переданный объект является экземпляром класса Employee. (instanceOf)
+//     getEmployees() - возвращает массив всех сотрудников.
+// 8. Проверьте корректную работу:
+//     const company = new Company("Tech Corp", "123-456", "Main Street");
+//     const emp1 = new Employee("John", "Doe", "Developer", 3000);
+//     const emp2 = new Employee("Barbara", "Johnson", "QA", 2500);
+//     company.addEmployee(emp1);
+//     company.addEmployee(emp2);
+//     console.log(company.getEmployees()); // [Employee, Employee]
+// 9. Добавьте в класс Company метод getInfo(), который возвращает строку с информацией о компании вида (перенос строки сделать с \n):
+// Компания: 
+// Адрес:
+// Количество сотрудников:
+
+//Task 2
+// 1. Реализуйте метод findEmployeeByName(firstName: string) в классе Company, который возвращает объект сотрудника, если такой найден.
+// 2. Реализуйте метод removeEmployee(firstName) в классе Company, который удаляет сотрудника по имени. 
+//    Метод должен искать сотрудника по имени и, если сотрудник найден, удалять его из массива.
+//    Для корретной работы создайте дополнительно приватный метод getEmployeeIndex(firstName), и используйте его в removeEmployee
+// 3. Добавьте метод getTotalSalary() в классе Company, который возвращает сумму всех зарплат сотрудников.
+
+// 4. Добавление валидации для полей сотрудника в сеттеры:
+//   Employee:
+//     firstName и lastName — строка от 2 до 50 символов, только латинские буквы.
+//     profession — строка, не может быть пустой, только латинские буквы и пробелы.
+//     salary — число, должно быть больше 0 и меньше 10000.
+// 5. Проверьте свой код:
+//     const emp1 = new Employee("John", "Doe", "Developer", 3000);
+//     const emp2 = new Employee("Jane", "Smith", "Manager", 5000);
+//     const emp3 = new Employee("Mark", "Brown", "Designer", 4000);
+
+//     const company = new Company("Tech Corp", "123-456", "Main Street");
+//     company.addEmployee(emp1);
+//     company.addEmployee(emp2);
+//     company.addEmployee(emp3);
+
+//     console.log(company.getTotalSalary()); // 12000
+//     console.log(company.findEmployeeByName("Jane")); // Employee { firstName: 'Jane', ... }
+//     company.removeEmployee("John");
+//     console.log(company.getEmployees()); // [Employee, Employee]
+
+
+
+
+class Employee {
+    #salary
+
+    constructor (firstName, lastName, profession, salary) {
+        this.firstName = firstName;
+        this.lastName = lastName;
+        this.profession = profession;
+        this.#salary = salary
+    }
+
+    get firstName() {
+        return this._firstName
+    }
+    get lastName() {
+        return this._lastName
+    }
+    get profession() {
+        return this._profession
+    }
+    get salary() {
+        return this.#salary
+    }
+
+    set firstName(firstName) {
+        return this._firstName = firstName;
+    }
```

Comment:
Для получения значения используем `get`
```suggestion
    set firstName(firstName) {
       this._firstName = firstName;
    }
```